core/shared/src/main/scala/scalaz/zio/TaskR.scala

```diff
@@ -0,0 +1,566 @@
+   * stack. Manual use of this method can improve fairness, at the cost of
+   * overhead.
+   */
+  final val yieldNow: UIO[Unit] = ZIO.Yield
+
+  /** Returns an effect that forks all of the specified values, and returns a
+   * composite fiber that produces a list of their results, in order.
+   */
+  final def forkAll[R, A](as: Iterable[ZIO[R, TaskrE, A]]): ZIO[R, Nothing, Fiber[TaskrE, List[A]]] =
+    ZIO.forkAll(as)
+
+  /** Returns an effect that forks all of the specified values, and returns a
+   * composite fiber that produces unit. This version is faster than [[forkAll]]
+   * in cases where the results of the forked fibers are not needed.
+   */
+  final def forkAll_[R, A](as: Iterable[ZIO[R, TaskrE, A]]): ZIO[R, Nothing, Unit] =
+    ZIO.forkAll_(as)
+
+  /** Returns an effect from a [[scalaz.zio.Exit]] value.
+   */
+  final def done[A](r: Exit[TaskrE, A]): IO[TaskrE, A] = ZIO.done(r)
+
+  /** Enables supervision for this effect. This will cause fibers forked by
+   * this effect to be tracked and will enable their inspection via [[ZIO.children]].
+   */
+  final def supervised[R, A](zio: ZIO[R, TaskrE, A]): ZIO[R, TaskrE, A] =
+    ZIO.supervised(zio)
+
+  /** Returns an effect that supervises the specified effect, ensuring that all
+   * fibers that it forks are interrupted as soon as the supervised effect
+   * completes.
+   */
+  final def supervise[R, A](zio: ZIO[R, TaskrE, A]): ZIO[R, TaskrE, A] =
+    ZIO.supervise(zio)
+
+  /** Returns an effect that supervises the specified effect, ensuring that all
+   * fibers that it forks are passed to the specified supervisor as soon as the
+   * supervised effect completes.
+   */
+  final def superviseWith[R, A](
+    zio: ZIO[R, TaskrE, A]
+  )(supervisor: IndexedSeq[Fiber[_, _]] => ZIO[R, Nothing, _]): ZIO[R, TaskrE, A] =
+    ZIO.superviseWith(zio)(supervisor)
+
+  /** Returns an effect that first executes the outer effect, and then executes
+   * the inner effect, returning the value from the inner effect, and effectively
+   * flattening a nested effect.
+   */
+  final def flatten[R, A](zio: ZIO[R, TaskrE, ZIO[R, TaskrE, A]]): ZIO[R, TaskrE, A] =
+    ZIO.flatten(zio)
+
+  /** Returns a lazily constructed effect, whose construction may itself require
+   * effects. This is a shortcut for `flatten(effectTotal(io))`.
+   */
+  final def suspend[R, A](io: => ZIO[R, TaskrE, A]): ZIO[R, TaskrE, A] =
+    ZIO.suspend(io)
+
+  /** Returns an effect that will execute the specified effect fully on the
+   * provided executor, before returning to the default executor.
+   */
+  final def lock[R, A](executor: Executor)(zio: ZIO[R, TaskrE, A]): ZIO[R, TaskrE, A] =
+    ZIO.lock(executor)(zio)
+
+  /** Imports an asynchronous effect into a pure `ZIO` value. See `effectAsyncMaybe` for
+   * the more expressive variant of this function that can return a value
+   * synchronously.
+   */
+  final def effectAsync[A](register: (ZIO[Any, TaskrE, A] => Unit) => Unit): ZIO[Any, TaskrE, A] =
```

Comment:
In the rare situation where `ZIO[Any, TaskrE, ...]` appears, that would translate to `Task[A]`.